quinn-proto/src/connection/streams/mod.rs

```diff
@@ -367,6 +385,28 @@ impl<'a> SendStream<'a> {
     }
 }
 
+struct WriteGuard<'a> {
+    limit: usize,
+    id: StreamId,
+    stream: &'a mut Send,
+    data_sent: &'a mut u64,
+    unacked_data: &'a mut u64,
+    pending: &'a mut PendingStreamsQueue,
```

Comment:
Should these be a single reference to a helper struct? That might also be easier to adapt to the `quinn` layer.